floodviz/peak_flow_utils.py

```diff
@@ -1,104 +1,125 @@
+    :param site: Site number on which to get data
+    :param date: Date for which to get the peak flow
+    :param url_prefix: NWIS service endpoint
+    :return: List of dictionaries containing the date and value of peak streamflow
+    """
+    url_prefix += 'dv/'
+    params = {
+        'format': 'rdb',
+        'siteStatus': 'all',
+        'sites': site,
+        'startDT': date,
+        'endDT': date
 
+    }
+    content = utils.parse_rdb(url_prefix, params)
+    if content is None:
+        print('rdb parser has returned no data to peakflow util req_peak_dv_data')
+        peaks = None
 
+    else:
+        # Find the name of that column whose name changes
+        for k in content[0].keys():
+            colname = re.match(r'\d+_00060_00003$', k)
+            # if a match has been found
+            if colname is not None:
+                colname = colname.string
+                break
+        # (no break) means that this column has not been found
+        else:
+            # I'd rather base my guess on what the thing looks like than on where it is
+            colname = content[0].keys()[3]
 
-def parse_peak_data(peak_data, dv_data):
+        keep_keys = [
+            colname,
+            'datetime'
+        ]
+        peaks = []
+        for raw in content:
+            peak = {k: raw[k] for k in keep_keys}
+            # rename that weird column to 'discharge'
+            peak['discharge'] = peak.pop(colname)
+            peaks.append(peak)
+
+    return peaks
 
-    """ 
-    Parses peak flow water data peak_data and constructs a dictionary 
-    appropriately formated for D3 charting library. 
 
-    ARGS: 
-        peak_data - list of lines from peak flow data requested from NWIS waterdata service.
-        dv_data - list of lines from daily value data requested from NWIS waterdata service.
-    
-    RETURNS:
-        peak_data - A list holding the peak flow data points
-        (each as a dict) for a specific site 
+def parse_peak_data(peak_data, dv_data):
+    """
+    Combine the peakflow information from the annual and daily values into a single list of dictionaries
+    describing yearly peak streamflow.
+    :param peak_data: Data from the NWIS peak flow service
+    :param dv_data: Data from the NWIS daily values service
+    :return: Data from peak flow service and daily value service combined into a single list.
     """
-    
     all_data = []
-    seen = set([])
+    seen = set()
     if peak_data:
-        # parse peak_data 
-        for line in peak_data:
-            if not line.startswith('USGS'):
-                continue
-            line = line.split('\t')
-            year = line[2].split('-')[0]
-            # remove duplicate years
-            if year in seen:
-                continue
-            else:
+        for point in peak_data:
+            # I am not worried about years that are not four digits long
+            year = re.match(r'^(\d{4}).*', point['peak_dt']).group(1)
```

Comment:
Not sure if we need this or not, it doesn't hurt to have this check I suppose but I didn't encounter any years that were not 4 digits.